nozl/schema/schema.go

```diff
@@ -30,42 +35,122 @@ func ParseOpenApiV3Schema(serviceID string, specFile []byte) error {
 		return err
 	}
 
-	var schemaList map[string]Schema
-	schemaKv, _ := eventstream.Eventstream.RetreiveKeyValStore(shared.ServiceKV)
-	entry, err := schemaKv.Get(serviceID)
+	for pathKey, pathValue := range doc.Paths {
 
-	if err != nil {
-		schemaList = make(map[string]Schema)
-	} else {
-		err := json.Unmarshal(entry.Value(), &schemaList)
-		if err != nil {
-			return err
+		if pathValue.Get != nil {
+			AddSchemaToKVStore(serviceID, pathKey, "get", pathValue.Get)
+		}
+		if pathValue.Post != nil {
+			AddSchemaToKVStore(serviceID, pathKey, "post", pathValue.Post)
+		}
+		if pathValue.Put != nil {
+			AddSchemaToKVStore(serviceID, pathKey, "put", pathValue.Put)
+		}
+		if pathValue.Patch != nil {
+			AddSchemaToKVStore(serviceID, pathKey, "patch", pathValue.Patch)
+		}
+		if pathValue.Delete != nil {
+			AddSchemaToKVStore(serviceID, pathKey, "delete", pathValue.Delete)
 		}
 	}
 
-	for pathKey, pathValue := range doc.Paths {
+	return nil
+}
 
-		if pathValue.Get != nil {
-			schemaList[pathValue.Get.OperationID] = addSchema(pathKey, "get", pathValue.Get)
-		} else if pathValue.Post != nil {
-			schemaList[pathValue.Post.OperationID] = addSchema(pathKey, "post", pathValue.Post)
-		} else if pathValue.Put != nil {
-			schemaList[pathValue.Put.OperationID] = addSchema(pathKey, "put", pathValue.Put)
-		} else if pathValue.Patch != nil {
-			schemaList[pathValue.Patch.OperationID] = addSchema(pathKey, "patch", pathValue.Patch)
-		} else if pathValue.Delete != nil {
-			schemaList[pathValue.Delete.OperationID] = addSchema(pathKey, "delete", pathValue.Delete)
+func MakeOptionalFieldsNullable(operation *openapi3.Operation) error {
+	if operation.RequestBody != nil {
+		ReqBodySchema := operation.RequestBody.Value.Content["application/x-www-form-urlencoded"].Schema.Value
```

Comment:
@mahad-akhter This header may break in other OAPI specs. Add a warning here at least.

Reply:
By warning, you mean a ToDo to take care of this later?

Reply:
Done